File: api/alerts.js
import { readFileSync } from "fs";
import { join } from "path";

export default function handler(req, res) {
  res.setHeader("Access-Control-Allow-Origin", "*");
  if (req.method === "OPTIONS") return res.status(200).end();

  try {
    const file = join(process.cwd(), "api", "data", "projects.json");
    const projects = JSON.parse(readFileSync(file, "utf8"));
    const alerts = projects
      .filter((p) => p.risk_category === "Critical" || p.risk_category === "High")
      .sort((a, b) => (b.risk_score || 0) - (a.risk_score || 0));
    res.status(200).json(alerts);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
}

File: backend/app/routers/alerts.py
from fastapi import APIRouter

from app.data.loader import load_projects

router = APIRouter(prefix="/api/alerts", tags=["alerts"])


@router.get("")
def alerts():
    projects = load_projects()
    high_risk = [p for p in projects if p["risk_category"] in ("High", "Critical")]
    high_risk.sort(key=lambda p: p["risk_score"], reverse=True)
    return high_risk
